dot/types/header.go

```diff
@@ -68,7 +68,11 @@ func (bh *Header) DeepCopy() (*Header, error) {
 	if len(bh.Digest.Types) > 0 {
 		cp.Digest = NewDigest()
 		for _, d := range bh.Digest.Types {
-			err := cp.Digest.Add(d.Value())
+			digestValue, err := d.Value()
+			if err != nil {
+				return nil, err
```

Comment:
nit (also adding the digest type index would be nice)

```suggestion
				return nil, fmt.Errorf("getting digest type value: %w", err)
```